temporal/api/cloud/nexus/v1/message.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package temporal.api.cloud.nexus.v1;
+
+option go_package = "go.temporal.io/api/cloud/nexus/v1;nexus";
+
+import "google/protobuf/timestamp.proto";
+
+message EndpointSpec {
+    // The name of the endpoint. Must be unique within an account.
+    // The name must match `^[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9]$`.
+    // This field is mutable.
+    string name = 1;
+
+    // Indicates where the endpoint should forward received nexus requests to. 
+    EndpointTargetSpec target_spec = 2;
+
+    // The set of policies (e.g. authorization) for the endpoint. Each request's caller
+    // must match with at least one of the specs to be accepted by the endpoint.
+    // This field is mutable.
+    repeated EndpointPolicySpec policy_specs = 3;
+
+    // The markdown description of the endpoint - optional.
+    string description = 4;
+}
+
+message EndpointTargetSpec {
+    // A target spec for routing nexus requests to a specific cloud namespace worker.
+    WorkerTargetSpec worker_target_spec = 1;
+}
+
+message WorkerTargetSpec {
+    // The target cloud namespace to route requests to. Namespace must be in same account as the endpoint. This field is mutable.
+    string namespace_id = 1;
+
+    // The task queue on the cloud namespace to route requests to. This field is mutable.
+    string task_queue = 2;
+}
+
+message EndpointPolicySpec {
+    // A policy spec that allows one caller namespace to access the endpoint. 
+    AllowedCloudNamespacePolicySpec allowed_cloud_namespace_policy_spec = 1;
+}
+
+message AllowedCloudNamespacePolicySpec {
+    // The namespace that is allowed to call into this endpoint. Calling namespace must be in same account as the endpoint.
+    string namespace_id = 1;
+}
+
+// An endpoint that receives and then routes Nexus requests
+message Endpoint {
+    // The id of the endpoint. This is generated by the server and is immutable.
+    string id = 1;
+
+    // The current version of the endpoint specification.
+    // The next update operation must include this version.
+    string resource_version = 2;
+
+    // The endpoint specification.
+    EndpointSpec spec = 3;
+
+    // The current state of the endpoint.
+    // Possible values: activating, activationfailed, active, updating, updatefailed, deleting, deletefailed, deleted
+    // For any failed state, reach out to Temporal Cloud support for remediation.
+    string state = 4;
```

Comment:
Lets move to using an enum here. 
@nikki-dag

Reply:
We should reuse this: https://github.com/temporalio/api-cloud/pull/36/files#diff-b38afecfa7fefa137fb5dbe446fd43bd6603676748e6f061fcf1f03d2e386706R13